src/lib/phdi-building-blocks/tests/test_schemas.py

```diff
@@ -0,0 +1,247 @@
+
+    schema = yaml.safe_load(
+        open(pathlib.Path(__file__).parent / "assets" / "test_schema.yaml")
+    )
+
+    output_path = mock.Mock()
+    output_path.__truediv__ = (  # Redefine division operator to prevent failure.
+        lambda x, y: x
+    )
+    output_format = "parquet"
+    fhir_url = "some_fhir_server_url"
+    access_token = "some_access_token"
+
+    fhir_server_responses = json.load(
+        open(
+            pathlib.Path(__file__).parent
+            / "assets"
+            / "FHIR_server_query_response_200_example.json"
+        )
+    )
+
+    query_response_1 = mock.Mock()
+    query_response_1.status_code = fhir_server_responses["status_code_1"]
+    query_response_1.content = json.dumps(fhir_server_responses["content_1"])
+
+    query_response_2 = mock.Mock()
+    query_response_2.status_code = fhir_server_responses["status_code_2"]
+    query_response_2.content = json.dumps(fhir_server_responses["content_2"])
+
+    patch_query.side_effect = [query_response_1, query_response_2]
+
+    make_table(
+        schema["my_table"],
+        output_path,
+        output_format,
+        fhir_url,
+        access_token,
+    )
+
+    assert len(patch_write.call_args_list[0]) == 2
+
+    assert patch_write.call_args_list[0][0] == (
+        [
+            apply_schema_to_resource(
+                fhir_server_responses["content_1"]["entry"][0]["resource"],
+                schema["my_table"],
+            )
+        ],
+        output_path,
+        output_format,
+        None,
+    )
+    assert patch_write.call_args_list[1][0] == (
+        [
+            apply_schema_to_resource(
+                fhir_server_responses["content_2"]["entry"][0]["resource"],
+                schema["my_table"],
+            )
+        ],
+        output_path,
+        output_format,
+        patch_write(
+            (
+                [
+                    apply_schema_to_resource(
+                        fhir_server_responses["content_1"]["entry"][0]["resource"],
+                        schema["my_table"],
+                    )
+                ],
+                output_path,
+                output_format,
+                None,
+            )
+        ),
+    )
+
+
+@mock.patch("phdi_building_blocks.schemas.write_schema_table")
+@mock.patch("phdi_building_blocks.schemas.fhir_server_get")
+def test_make_table_fail(patch_query, patch_write):
+
+    schema = {}
+
+    output_path = mock.Mock()
+    output_path.__truediv__ = (  # Redefine division operator to prevent failure.
+        lambda x, y: x
+    )
+
+    output_format = "parquet"
+
+    fhir_url = "some_fhir_server_url"
+    access_token = "some_access_token"
+
+    response = mock.Mock()
+    response.status_code = 400
+    patch_query.return_value = response
+
+    make_table(
+        schema,
+        output_path,
+        output_format,
+        fhir_url,
+        access_token,
+    )
+
+    assert len(patch_write.call_args_list) == 0
```

Comment:
I'm not quite sure I understand how this assertion tests that `make_table` is working as expected when it fails. Can you elaborate on this a bit?

Reply:
This is an overly complicated way to assert that the function wasn't called. Since writing this I learned about `assert_not_called()` and used it elsewhere in these tests. It would probably be a clearer choice here. The goal of this test was to ensure that given a `status_code != 200` we don't run any of the code in that block.

Reply:
[25fbc25](https://github.com/CDCgov/prime-public-health-data-infrastructure/pull/129/commits/25fbc2543798f8861ab233cb5f665e02eaebb030) makes the change to `assert_not_called()`.